eth2util/deposit/deposit.go

```diff
@@ -241,3 +244,61 @@ func EthsToGweis(ethAmounts []int) []eth2p0.Gwei {
 
 	return gweiAmounts
 }
+
+// WriteClusterDepositDataFiles writes deposit-data-*eth.json files for each distinct amount.
+func WriteClusterDepositDataFiles(depositDatas [][]eth2p0.DepositData, network string, clusterDir string, numNodes int) error {
+	// The loop across partial amounts (shall be unique)
+	for _, dd := range depositDatas {
+		for n := 0; n < numNodes; n++ {
+			nodeDir := path.Join(clusterDir, fmt.Sprintf("node%d", n))
+			if err := WriteDepositDataFile(dd, network, nodeDir); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+// WriteDepositDataFile writes deposit-data-*eth.json file for the provided depositDatas.
+// The amount will be reflected in the filename in ETH.
+// All depositDatas amounts shall have equal values.
```

Comment:
Not in this case. This function writes the single deposit file for a node. In case of partial deposits this function is called multiple times per each partial amount x each node. The input slice here is the input for the single file. All amounts shall be equal.